### PR TITLE
Fix/span

### PR DESCRIPTION
**Pull request by h-2**

### CODE REVIEW

File: include/seqan3/std/span [hunk 1]
```diff
@@ -253,137 +253,137 @@ class span<span_tp, dynamic_extent> {
...
                           nullptr_t> = nullptr) noexcept
-        : data{other.get_data()}, size{other.get_size()} {}
+        : data_{other.data()}, size_{other.size()} {}
 
 //    ~span() noexcept = default;
 
     template <ptrdiff_t count>
     inline constexpr span<element_type, count> first() const noexcept
     {
-        static_assert(count >= 0, "Count must be >= 0 in span::first()");
```
Comment (eseiler) [inline include/seqan3/std/span]: This is in an constexpr context, so it could be `static_asserts` (?)
So why change it?

Comment (h-2) [reply include/seqan3/std/span]: i was a little over-ambitious with removing the wrong asserts maybe

**Comment by h-2** [comment]
> I guess changing all static_asserts to asserts is fine since the STL documents out of range errors to be undefined behaviour.

No, those were actual code errors in LLVM's STL, you cannot static assert on `data()`, even if `data()` is constexpr, because `data()` is not always evaluated in a constexpr-context.

**Comment by smehringer** [comment]
Well then next time a comment in the PR would be very helpful for the review :pray:

**Comment by h-2** [comment]
> Well then next time a comment in the PR would be very helpful for the review pray

:+1:

**Pull request merged**